project/source/shared/SceneNavigationCamera.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "Ecs.h"
+#include "Input.h"
+
+namespace nc::sample
+{
+    /** Control camera pan, look, and zoom with right mouse button, middle mouse button, and mouse wheel. */
+    class SceneNavigationCamera : public Camera
+    {
+        public:
+            SceneNavigationCamera(EntityHandle handle);
```

Comment:
It would be nice to make this configurable such that you can change the origin to be the selected object. (Rotation around some selected object)